ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -436,15 +441,21 @@ public boolean maybeFailedDueToOfflineReplicas() {
       routerMetrics.failedMaybeDueToOriginatingDcOfflineReplicasCount.inc();
       return true;
     }
-    if (hasFailedOnCrossColoNotFound() && allReplicaCount - totalNotFoundCount >= replicaSuccessTarget
-        && totalOfflineReplicaCount > 0) {
+
+    if (hasFailedOnCrossColoNotFound()
+        && allReplicaCount - totalNotFoundCount >= replicaSuccessTarget
+        && (totalOfflineReplicaCount > 0
+            || (originatingDcEligibleReplicaCount < originatingDcTotalReplicaCount - routerConfig.routerPutSuccessTarget + 1))
```

Comment:
Can you improve the readability of this condition by decomposing it into smaller statements?

Something like:

final boolean isEnoughReplicaOnline = allReplicaCount - totalNotFoundCount >= replicaSuccessTarget;
.
.
if (hasFailedOnCrossColoNotFound() && isEnoughReplicaOnline ...)

Reply:
The new revision added comment.